http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -815,11 +818,11 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       nodepool = saveClusterResult.defaultNodepool.toNodepool()
 
       // Retrieve a pet identity from Sam
-      runtimeServiceAccountOpt <- serviceAccountProvider.getClusterServiceAccount(userInfo, cloudContext)
-      _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done Sam call for getClusterServiceAccount")))
-      petSA <- F.fromEither(
-        runtimeServiceAccountOpt.toRight(new Exception(s"user ${userInfo.userEmail.value} doesn't have a PET SA"))
-      )
+      petSA <- cloudContext match {
+        case CloudContext.Gcp(googleProject)       => samService.getPetServiceAccount(userInfo, googleProject)
```

Comment:
I am wondering if this branching logic should be abstracted away in the samService, something like `getPet`? What do you think? It would make things more DRY, but at the expense of some duck typing 🤔

Reply:
Good point - the old `SamAuthProvider` class had a method like this. It uses the same types for GCP and Azure so I can pull it into `SamService`, good call.